pkg/feeder/feeder.go

```diff
@@ -143,6 +153,11 @@ func (c *Client) newRequest(method string, path string, query map[string]string,
 // do executes a request and waits for response and returns an error
 // otherwise.
 func (c *Client) do(req *http.Request, v any) (*http.Response, error) {
+	<-c.available
+	defer func() {
+		c.available <- true
+	}()
+
```

Comment:
If I understand well, the `available` channel is for limiting and managing the requests cadence. I suggest using a channel of `*http.Request` (instead of `bool`), then you can use all the behavior of channels to manage the request flow (lock if the channel is full or add the request to the channel queue on another case). If the channel will be of size 1, then I suggest using a mutex instead channels.

Reply:
The idea behind using channels instead mutex is that the mutex is going to freeze the entire app, while the channel will only freeze the goroutines that are fetching information. On the other hand, this keeps simple the change and don't introduce any big bug while changing the channel to `{res, req}` can be a huge source of problems and that's not the scope of this issue, later in the future might be useful to have a refactorization around the feeder gateway.